src/Testcontainers.Db2/Db2Builder.cs

```diff
@@ -0,0 +1,202 @@
+            .WithEnvironment("DB2INSTANCE", username);
+    }
+
+    /// <summary>
+    /// Sets the Db2 password.
+    /// </summary>
+    /// <param name="password">The Db2 password.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(password: password))
+            .WithEnvironment("DB2INST1_PASSWORD", password);
+    }
+
+    /// <summary>
+    /// Sets the Db2 archive logs.
+    /// </summary>
+    /// <param name="archiveLogs">The Db2 archive logs setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithArchiveLogs(bool archiveLogs)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(archiveLogs: archiveLogs))
+            .WithEnvironment("ARCHIVE_LOGS", archiveLogs.ToString());
+    }
+
+    /// <summary>
+    /// Sets the Db2 autoconfig setting.
+    /// </summary>
+    /// <param name="autoConfig">The Db2 autoconfig setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithAutoconfig(bool autoConfig)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(autoConfig: autoConfig))
+            .WithEnvironment("AUTOCONFIG", autoConfig.ToString());
+    }
+
+    /// <summary>
+    /// Accepts the Db2 license agreement.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithLicenseAgreement()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultLicenseAgreement))
+            .WithEnvironment("LICENSE", DefaultLicenseAgreement);
+    }
+
+    /// <summary>
+    /// Maps the database to memory.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithInMemoryDatabase()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultInMemoryDatabasePath))
+            .WithTmpfsMount(DefaultInMemoryDatabasePath);
+    }
+
+    /// <inheritdoc />
+    public override Db2Container Build()
+    {
+        Validate();
+
+        // By default, the base builder waits until the container is running. However, for Db2, a more advanced waiting strategy is necessary
+        // If the user does not provide a custom waiting strategy, append the default Db2 waiting strategy.
+        var db2Builder = DockerResourceConfiguration.WaitStrategies.Count() > 1
+            ? this
+            : WithWaitStrategy(Wait.ForUnixContainer()
+                .UntilMessageIsLogged("All databases are now active")
+                .UntilMessageIsLogged("Setup has completed.")
+                .AddCustomWaitStrategy(new WaitUntil(DockerResourceConfiguration))
+            );
```

Comment:
Please move the wait strategy configuration to `Init()`. We do not need the pattern here. We only need it if we need to configure the wait strategy with developer-provided configurations.